src/Shared/UnitTests/ObjectModelHelpers.cs

```diff
@@ -155,6 +155,36 @@ internal static void AssertItemEvaluationFromGenericItemEvaluator(Func<string, P
             }
         }
 
+        internal static void ShouldHaveSucceeded(this BuildResult result)
+        {
+            result.OverallResult.ShouldBe(BuildResultCode.Success, result.Exception is not null ? result.Exception.ToString() : string.Empty);
+        }
+
+        internal static void ShouldHaveSucceeded(this GraphBuildResult result)
+        {
+            result.OverallResult.ShouldBe(BuildResultCode.Success, result.Exception is not null ? result.Exception.ToString() : string.Empty);
```

Comment:
What's the second half of this about an exception? If it succeeded, shouldn't there be no exception?

Reply:
The second argument is an info message that Shouldly prints out if the assertion fails. In this case, if the BuildResultCode is failure AND there's an exception, then I print the exception to make debugging easier. But if there's no exception there's no helpful message to write.

Reply:
Ah, makes sense, thanks!